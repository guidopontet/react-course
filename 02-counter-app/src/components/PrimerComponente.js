import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const PrimerComponente = ({ greeting, paragraph }) => {
  return (
    <Fragment>
      <h1>{ greeting }</h1>
      <p>{ paragraph }</p>
    </Fragment>
  );
};

PrimerComponente.prototypes = {
  greeting: PropTypes.string,
};

PrimerComponente.defaultProps = {
  paragraph: 'Es un p√°rrafo',
};

export default PrimerComponente;
