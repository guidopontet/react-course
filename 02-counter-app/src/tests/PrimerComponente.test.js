// import React from 'react';
// import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import { shallow } from 'enzyme';

import PrimerComponente from "../components/PrimerComponente";

describe('PrimerComponente', () => {
  // test('should show greeting message', () => {
  //   const greeting = 'Hola Crack,';

  //   const screen = render(<PrimerComponente greeting='Hola Crack,' />);

  //   expect(screen.getByText(greeting)).toBeInTheDocument();
  //  });

  test('should show PrimerComponente', () => {
    const greeting = 'Hola Crack,';
    const wrapper = shallow(<PrimerComponente greeting='Hola Crack,'/>);

    expect(wrapper).toMatchSnapshot();
  });

  test('should render paragraph by argument', () => {
    const greeting = 'Hola Crack,';
    const paragraph = 'Es un p√°rrafo';
    const wrapper = shallow(
      <PrimerComponente
        greeting={ greeting }
        paragraph={ paragraph }/>
    );

    const paragraphText = wrapper.find('p').text();

    expect(paragraphText).toBe(paragraph);
  });
 });
