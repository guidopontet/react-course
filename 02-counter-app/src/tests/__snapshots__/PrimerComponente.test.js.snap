// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimerComponente should show PrimerComponente 1`] = `
<Fragment>
  <h1>
    Hola Crack,
  </h1>
  <p>
    Es un p√°rrafo
  </p>
</Fragment>
`;
